app/assets/javascripts/index/tileinfo.js

```diff
@@ -0,0 +1,94 @@
+OSM.TileInfo = function(map) {
+  var tile;
+
+  function lon2x(lon, zoom) {
+    return Math.floor((lon + 180) / 360 * Math.pow(2, zoom));
+  }
+
+  function lat2y(lat, zoom) {
+    return Math.floor((1 - Math.log(Math.tan(lat * Math.PI / 180) + 1 /
+                                    Math.cos(lat * Math.PI / 180)) / Math.PI) / 2 * Math.pow(2, zoom));
+  }
+
+  function x2lon(x, z) {
+    return x / Math.pow(2, z) * 360 - 180;
+  }
+
+  function y2lat(y, z) {
+    var n = Math.PI - 2 * Math.PI * y / Math.pow(2, z);
+    return 180 / Math.PI * Math.atan(0.5 * (Math.exp(n) - Math.exp(-n)));
+  }
+
+  var page = {};
+
+  page.pushstate = page.popstate = function(path) {
+    OSM.loadSidebarContent(path, function () {
+      page.load(path, true);
+    });
+  };
+
+  page.load = function(path, noCentre) {
+    var params = querystring.parse(path.substring(path.indexOf('?') + 1)),
+        lat = parseFloat(params.lat),
+        lon = parseFloat(params.lon),
+        zoom = parseInt(params.zoom),
+        y = lat2y(lat, zoom),
+        x = lon2x(lon, zoom),
+        sw = L.latLng(y2lat(y, zoom), x2lon(x, zoom)),
+        ne = L.latLng(y2lat(y + 1, zoom), x2lon(x + 1, zoom)),
+        bounds = L.latLngBounds(sw, ne),
+        latlng = L.latLng(lat, lon),
+        layer = map.getBaseLayer(),
+        url = layer.getTileUrl({ x: x, y : y });
+
+    tile = L.rectangle(bounds, {
+      color: "#FF6200",
+      weight: 4,
+      opacity: 1,
+      interactive: false
+    }).addTo(map);
+
+    $("#tileinfo-url").attr("href", url).text(url);
+    $("#tileinfo-z").text(zoom);
+    $("#tileinfo-y").text(y);
+    $("#tileinfo-x").text(x);
+
+    if (layer.options.code === "M" || layer.options.code === "H")
+    {
+      $.ajax({
+        url: url + "/status",
+        success: function (data) {
+          var re = /^Tile is (.*?)\. Last rendered at (.*?)\. Last accessed at (.*?)\./,
+              matches = data.match(re);
+
+          if (matches)
+          {
+            var state = matches[1],
+                rendered = Date.parse(matches[2]),
+                accessed = Date.parse(matches[3]);
+
+            $("#tileinfo-state").text(I18n.t("javascripts.tileinfo.state." + state));
+            $("#tileinfo-rendered").text(I18n.l("time.formats.friendly", rendered));
+            $("#tileinfo-accessed").text(I18n.l("time.formats.friendly", accessed));
+          }
+        }
+      });
+    }
+    else
+    {
+      $("#tileinfo-status").hide();
```

Comment:
Well the page will have been reloaded by then won't it? That will only be an issue if we start updating the details on a layer change.